### PR TITLE
New repo format. Big changes

### PR DESCRIPTION
**Pull request by vikmeup**
PR brings many changes:

1. Moved repo tests to Jest and converted most of the `js` scripts to `ts`. Fives better structure to write test with Jest.
2. From now on assets folder contains only checksum addresses, no more lowercased
3. Added script to generate `whitelist` and `blacklist` based on assets already added to repo and assured address from bl not present in wl and apposite direction.
4. Added more test to verify repo structured in proper way, and added more tests
5. Added multiple scripts to checksum address for ethreum's, bep2.

Classes: https://github.com/trustwallet/assets/issues/291, https://github.com/trustwallet/assets/issues/581

### CODE REVIEW

File: src/test/gen_list.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    writeFileSync,
+    sortDesc,
+    getUnique,
+    mapList
+} from './helpers'
+
+// Current chains where assets tokens supported by Trust Wallet
+// TODO auto create list based on assets folder presence
+const chainsSupportedWhiteAndBlacklist = [
+    'binance',
```
Comment (vikmeup) [inline src/test/gen_list.ts]: use `@trustwallet/types` lib to pull coin types

File: src/test/gen_list.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    getChainBlacklistPath,
+    writeFileSync,
+    sortDesc,
+    getUnique,
+    mapList
+} from './helpers'
+
+// Current chains where assets tokens supported by Trust Wallet
+// TODO auto create list based on assets folder presence
+const chainsSupportedWhiteAndBlacklist = [
```
Comment (vikmeup) [inline src/test/gen_list.ts]: did you consider having shorter names for variables?

Comment (vikmeup) [reply src/test/gen_list.ts]: can you remove comments? as you can explain with name variable

File: src/test/gen_list.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    'binance',
+    'classic',
+    'eos',
+    Ethereum,
```
Comment (dpereskokov) [inline src/test/gen_list.ts]: Lets make it consistent. Either all to consts or all as strings

File: src/test/gen_list.ts (hunk 2 above)
Comment (dpereskokov) [reply src/test/gen_list.ts]: Do we need it? Is it possible just to iterate over blockchains directory and process those containing assets directory inside?

File: src/test/gen_list.ts (hunk 3 above)
Comment (kolya182) [reply src/test/gen_list.ts]: https://github.com/trustwallet/assets/pull/502/commits/9e8ab2a3deb361b38deca959c91fbd5e46117478

File: src/test/gen_list.ts (hunk 2 above)
Comment (kolya182) [reply src/test/gen_list.ts]: https://github.com/trustwallet/assets/pull/502/commits/9e8ab2a3deb361b38deca959c91fbd5e46117478

File: src/test/gen_list.ts (hunk 1 above)
Comment (kolya182) [reply src/test/gen_list.ts]: https://github.com/trustwallet/assets/pull/502/commits/9e8ab2a3deb361b38deca959c91fbd5e46117478

File: src/test/gen_list.ts (hunk 2 above)
Comment (kolya182) [reply src/test/gen_list.ts]: Initial idea was to have list in case someone adds assets to new chain without tests

**Pull request merged**